docs/modules/ROOT/pages/extensions/freemarker.adoc

```diff
@@ -0,0 +1,53 @@
+[[freemarker]]
+= Freemarker Extension
+
+*Since Camel Quarkus 0.3.0*
+
+The Freemarker extension component allows processing a message using a FreeMarker template. This can be ideal when using Templating to generate responses for requests.
+
+Maven users will need to add the following dependency to their `pom.xml` for this extension.
+
+[source,xml]
+------------------------------------------------------------
+<dependency>
+    <groupId>org.apache.camel.quarkus</groupId>
+    <artifactId>camel-quarkus-freemarker</artifactId>
+</dependency>
+------------------------------------------------------------
+
+== Usage
+
+The extension provides support for the Camel https://camel.apache.org/components/latest/freemarker-component.html[Freemarker Component].
+
+=== Configuration
+
+The Camel Freemarker extension exposes the following configuration properties.
+
+[cols="80,.^10,.^10"]
+|===
+|Configuration property |Type |Default
+
+|`quarkus.camel.freemarker.locations`
+
+ Comma-separated list of locations to scan recursively for templates.
+|List<String>
+|`freemarker/templates`
+
+|`quarkus.camel.freemarker.include-patterns`
+
+ Used for inclusive filtering scanning of Freemarker templates. By default all tree files inside 'locations' will be included.
```

Comment:
Plase add:

* Where we scan for the templates (classpath, filesystem)
* Are `classpath:` and `filesystem:` prefixes supported? Which one is the default? (I do not insist on supporting `filesystem:`) We have other extensions documenting similar features properly, just grep for `classpath:` in the source tree.